src/MOIWrapper.jl

```diff
@@ -0,0 +1,558 @@
+   :Scaling => scaling,
+   :Perturbation => set_perturbation,
+   #:Algorithm => set_algorithm
+   )
+# These options are set by using the ClpSolve object
+const solveoptionmap = Dict(
+   :PresolveType => set_presolve_type,
+   :SolveType => set_solve_type,
+   :InfeasibleReturn => set_infeasible_return,
+   )
+
+function setoption(m::ClpOptimizer, name::Symbol, value)
+    if haskey(optionmap, name)
+        optionmap[name](m.inner,value)
+    elseif haskey(solveoptionmap, name)
+        solveoptionmap[name](m.solveroptions,value)
+    else
+        error("Unrecognized option: $name")
+    end
+end
+
+function ClpOptimizer(;kwargs...)
+    # env = Env()
+    m = ClpOptimizer(nothing)
+    # m.env = env
+    m.params = Dict{String,Any}()
+    MOI.empty!(m)
+    for (name,value) in kwargs
+        m.params[Symbol(name)] = value
+    end
+    return m
+end
+
+LQOI.LinearQuadraticModel(::Type{ClpOptimizer},env) = ClpModel() 
+
+LQOI.supported_constraints(s::ClpOptimizer) = SUPPORTED_CONSTRAINTS
+LQOI.supported_objectives(s::ClpOptimizer) = SUPPORTED_OBJECTIVES
+
+"""
+Change the bounds of the variable. The sense of the upperbound
+is given by `backend_type(m, Val{:Upperbound}())`. The sense
+of the lowerbound is given by `backend_type(m, Val{:Lowerbound}())`
+"""
+function LQOI.change_variable_bounds!(instance::ClpOptimizer, cols::Vector{Int}, values::Vector{Float64}, 
+        senses::Vector)
+    
+    upperbounds = replaceInf(get_col_upper(instance.inner))    
+    lowerbounds = replaceInf(get_col_lower(instance.inner))    
+    
+    for i in 1:length(senses)
+        if senses[i] == Cchar('U')
+            upperbounds[cols[i]] = values[i]
+        elseif senses[i] == Cchar('L')
+            lowerbounds[cols[i]] = values[i]
+        else            
+            error("sense is " * string(senses[i]) * ", but only " * Cchar('U') *  
+            " and " * Cchar('L') * " senses are supported")
+        end
+    end
+    
+    chg_column_upper(instance.inner, upperbounds)
+    chg_column_lower(instance.inner, lowerbounds)
+end
+
+"""
+get_variable_lowerbound(m, col::Int)::Float64
+
+Get the lower bound of the variable in 1-indexed column `col` of the model `m`.
+"""
+function LQOI.get_variable_lowerbound(instance::ClpOptimizer, col::Int)::Float64
+    lower = replaceInf(get_col_lower(instance.inner))
+    return lower[col];  
+end
+
+"""
+get_variable_upperbound(m, col::Int)::Float64
+
+Get the upper bound of the variable in 1-indexed column `col` of the model `m`.
+"""
+function LQOI.get_variable_upperbound(instance::ClpOptimizer, col::Int)::Float64
+    upper = replaceInf(get_col_upper(instance.inner))
+    return upper[col];  
+end
+
+"""
+get_number_linear_constraints(m)::Int
+
+Get the number of linear constraints in the model `m`.
+"""
+function LQOI.get_number_linear_constraints(instance::ClpOptimizer)
+    return get_num_rows(instance.inner)
+end
+
+function addrow(row, lower, upper, instance, rows, cols, coefs)
```

Comment:
The ordering on these arguments is inconsistent with the rest of the calls. They could be typed for readability?

Reply:
It is also pretty confusing that you pass in the A matrix but only add one row. At the very least it should be documented